include/seqan/basic/iterator_zipped.h

```diff
@@ -0,0 +1,497 @@
+class Iter<std::tuple<TIteratorPack...>, ZipIterator>
+{
+public:
+    std::tuple<TIteratorPack...> dataIter;  // tuple stores the different iterators.
+
+    // Default c'tor.
+    Iter()
+    {
+        static_assert(sizeof...(TIteratorPack) > 0, "Requires at least one argument!");
+    }
+
+    // Custom c'tor to create from tuple.
+    Iter(std::tuple<TIteratorPack...> && iterTuple) : dataIter(iterTuple)
+    {
+        static_assert(sizeof...(TIteratorPack) > 0, "Requires at least one argument!");
+    }
+};
+
+
+// ============================================================================
+// Metafunctions
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// Metafunction MakeTupleIndices
+// ----------------------------------------------------------------------------
+
+// This metafunction recursively calls itself, while adding a new index to the growing
+// Indices pack.
+template<unsigned ID, unsigned... Indices>
+struct MakeTupleIndices :
+MakeTupleIndices<ID-1, ID-1, Indices...>
+{};
+
+// Base case to stop the recursion and to define the TupleIndices with the indices pack.
+template<unsigned... Indices>
+struct MakeTupleIndices<0, Indices...>
+{
+    typedef TupleIndices<Indices...> Type;
+};
+
+// ----------------------------------------------------------------------------
+// Metafunction Value
+// ----------------------------------------------------------------------------
+
+// We expand each iterator type in the param pack to call its Value metafunction
+// and define a tuple with the value types.
+template <typename... TIteratorPack>
+struct Value<Iter<std::tuple<TIteratorPack...>, ZipIterator> >
+{
+    typedef std::tuple<typename Value<TIteratorPack>::Type...>  Type;
+};
+
+// ----------------------------------------------------------------------------
+// Metafunction GetValue
+// ----------------------------------------------------------------------------
+
+// See above.
+template <typename... TIteratorPack>
+struct GetValue<Iter<std::tuple<TIteratorPack...>, ZipIterator> >
+{
+    typedef std::tuple<typename GetValue<TIteratorPack>::Type...>  Type;
+};
+
+// ----------------------------------------------------------------------------
+// Metafunction Reference
+// ----------------------------------------------------------------------------
+
+// See above.
+template <typename... TIteratorPack>
+struct Reference<Iter<std::tuple<TIteratorPack...>, ZipIterator> >
+{
+    typedef std::tuple<typename Reference<TIteratorPack>::Type...>  Type;
+};
+
+// Unclear why we need that! It should be possible to let the generic implementation
+// in iterator_base.h to take care of it. Apparently, it is the same semantic.
+template <typename... TIteratorPack>
+struct Reference<Iter<std::tuple<TIteratorPack...>, ZipIterator> const> :
+    Reference<Iter<std::tuple<TIteratorPack...>, ZipIterator> > {};
+
```

Comment:
Nop. The reference is the reference to the value of the underlying container an iterator is pointing to. Hence the const qualification depends on the constness of the container not the iterator itself. 
However, there is a generic MF that should do the above thing but it does not work for whatever reasons.